Public/Src/Tools/Execution.Analyzer/Analyzers.Core/XLGPlusPlus/Events.proto

```diff
@@ -22,31 +34,845 @@ message EventTypeQuery{
+    ZwSetFileNameInformationFileSource = 18;
+    ZwSetFileNameInformationFileDest = 19;
+    CopyFileSource = 20;
+    CopyFileDestination = 21;
+    CreateHardLinkSource = 22;
+    CreateHardLinkDestination = 23;
+    RemoveDirectory = 24;
+    NtQueryDirectoryFile = 25;
+    ZwQueryDirectoryFile = 26;
+    NtCreateFile = 27;
+    ZwCreateFile = 28;
+    ZwOpenFile = 29;
+    ChangedReadWriteToReadAccess = 30;
+    FirstAllowWriteCheckInProcess = 31;
+    ReparsePointTarget = 32;
+    CreateSymbolicLinkSource = 33;
+    MoveFileWithProgressSource = 34;
+    MoveFileWithProgressDest = 35;
+    MultipleOperations = 36;
+    ProcessExit = 37;
+    MacLookup = 38;
+    MacReadlink = 39;
+    MacVNodeCreate = 40;
+    KAuthMoveSource = 41;
+    KAuthMoveDest = 42;
+    KAuthCreateHardlinkSource = 43;
+    KAuthCreateHardlinkDest = 44;
+    KAuthCopySource = 45;
+    KAuthCopyDest = 46;
+    KAuthDeleteDir = 47;
+    KAuthDeleteFile = 48;
+    KAuthOpenDir = 49;
+    KAuthReadFile = 50;
+    KAuthCreateDir = 51;
+    KAuthWriteFile = 52;
+    KAuthClose = 53;
+    KAuthCloseModified = 54;
+    KAuthGetAttributes = 55;
+    KAuthVNodeExecute = 56;
+    KAuthVNodeWrite = 57;
+    KAuthVNodeRead = 58;
+    KAuthVNodeProb = 59;
+}
+
+enum FingerprintComputationKind_XLGpp{
+    Execution = 0;
+    CacheCheck = 1;
+}
+
+enum ObservedInputType_XLGpp{
+    AbsentPathProbe = 0;
+    FileContentRead = 1;
+    DirectoryEnumeration = 2;
+    ExistingDirectoryProbe = 3;
+    ExistingFileProbe = 4;
+}
+
+enum PreserveOutputsMode_XLGpp{
+    Disabled = 0;
+    Enabled = 1;
+    Reset = 2;
+}
+
+enum SandboxKind_XLGpp{
+    SandboxKind_None = 0;
+    Default = 1;
+    WinDetours = 2;
+    MaxOsKext = 3;
+    MacOsKextIgnoreFileAccesses = 4;
+}
+
+enum DoubleWritePolicy_XLGpp{
+    DoubleWritesAreErrors = 0;
+    AllowSameContentDoubleWrites = 1;
+    UnsafeFirstDoubleWriteWins = 2;
+}
+
+message AbsolutePath_XLGpp{
+    string Value = 1;
+}
+
+message ContentHash_XLGpp{
+    string Value = 1;
+}
+
+message FileArtifact_XLGpp{
+    AbsolutePath_XLGpp path = 1;
+
+    int32 RewriteCount = 2;
+
+    bool IsSourceFile = 3; 
+
+    bool IsOutputFile = 4;
+}
+
+message FileContentInfo_XLGpp{
+    // [IsKnownExistenceFlag | Existence | IsKnownLengthFlag | Length]; 
+    int64 LengthAndExistence = 1;
+
+    ContentHash_XLGpp Hash = 2;
+}
+
+message PipExecutionPerformance_XLGpp{
+    int32 PipExecutionLevel = 3;
+
+    google.protobuf.Timestamp ExecutionStart = 4;
+    
+    google.protobuf.Timestamp ExecutionStop = 5;
+}
+
+message ProcessPipExecutionPerformance_XLGpp{
+    google.protobuf.Duration ProcessExecutionTime = 1;
+    
+    IOTypeCounters_XLGpp ReadCounters = 2;
+
+    IOTypeCounters_XLGpp WriteCounters = 3;
+
+    IOTypeCounters_XLGpp OtherCounters = 4;
+    
+    google.protobuf.Duration UserTime = 5;
+
+    google.protobuf.Duration KernelTime = 6;
+
+    uint64 PeakMemoryUsage = 7;
+
+    int32 PeakMemoryUsageMb = 8;
+
+    uint32 NumberOfProcesses = 9;

```

Comment:
why is one unsigned and the other one is signed?

Reply:
it is mimicing exactly what was in the corresponding BXL "ProcessPipExecutionPerformance" class

---
In reply to: [308895445](https://github.com/microsoft/BuildXL/pull/671#discussion_r308895445) [](ancestors = 308895445)